### PR TITLE
fix(button): hcm for ghost icon color contrast

### LINKED ISSUE #9903: [a11y]: Button: ghost kind in high contrast mode needs higher contrast ratio
### Package

carbon-components, carbon-components-react

### Browser

Chrome

### Operating System

Windows

### Package version

latest

### Automated testing tool and ruleset

Windows 10 High Contrast Mode

### Assistive technology

_No response_

### Description

The ghost style button needs to be easily visible in high contrast mode.

### WCAG 2.1 Violation

Success Criterion 1.4.3: Contrast (Minimum)

### CodeSandbox example

https://react.carbondesignsystem.com/?path=/story/components-button--playground
image

### Steps to reproduce

1. Turn on Windows 10 high contrast mode, [instructions](https://support.microsoft.com/en-us/windows/turn-high-contrast-mode-on-or-off-in-windows-909e9d89-a0f9-a3a9-b993-7a6dcee85025)
2. Visit https://react.carbondesignsystem.com/?path=/story/components-button--playground
3. Observe the ghost kind button is hard to see due to low color contrast

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

### PR DESCRIPTION
**Pull request by kinueng**
Closes #9903

Add CSS specifically for Windows high contrast mode for the ghost style `<Button>`

#### Changelog

**Changed**

Added CSS specific to the ghost styling for the Button that will apply `fill: white` to the SVG.

#### Testing / Reviewing

# Before fix
image
# After fix
image

### CODE REVIEW

**Review by abbeyhrt — APPROVED** [review]
Looks great to me! Thank you for the contribution! 🎉

File: packages/components/src/components/button/_button.scss [hunk 1]
```diff
@@ -285,6 +285,9 @@
     path:not([data-icon-path]):not([fill='none']),
   .#{$prefix}--btn--ghost.#{$prefix}--btn--icon-only .#{$prefix}--btn__icon {
     fill: $icon-01;
+    @include high-contrast-mode {
+      fill: $icon-03;
+    }
```
Comment (tay1orjones) [inline packages/components/src/components/button/_button.scss]: We can use the `icon-fill` parameter instead to ensure the fill is the same as the ButtonText forced color

https://github.com/carbon-design-system/carbon/blob/3c9fdbcc12116759c47605ad562aa4d3be78b9d5/packages/components/src/globals/scss/_helper-mixins.scss#L256-L258

```suggestion
    @include high-contrast-mode('icon-fill');
```

**Review by tay1orjones — CHANGES_REQUESTED** [review]
Thanks for submitting this fix! A couple things in addition to the change requested in this review below:

1. The HCM styles need to be at the very end of the file to ensure they're not overridden
2. This fix needs to be mirrored in ~`packages/carbon-react`~ -- Edit, sorry I meant `packages/styles`

**Review by tay1orjones — APPROVED** [review]
@kinueng I hope you don't mind - I went ahead and pushed an update to this PR with the suggestions above. Let me know what you think

**Comment by tay1orjones** [comment]
Fantastic! thanks @kinueng

**Pull request merged**